src/test/scala/com/fortysevendeg/scala/android/unit/models/FormValidationSpec.scala

```diff
@@ -0,0 +1,135 @@
+
+    "return an error when the name field is empty" in {
+
+      val valueToTest = ""
+      val editableField = mockEditText(valueToTest)
+
+      val form = FormModel(Some(editableField), None, None)
+
+      val result = isValidNameField(form)
+
+      result.isSuccess === false
+      result.toEither.left.get.details === s"name can not be empty."
+    }
+
+    "return success when the name field is specified" in {
+
+      val valueToTest = "John Doe"
+      val editableField = mockEditText(valueToTest)
+
+      val form = FormModel(Some(editableField), None, None)
+
+      val result = isValidNameField(form)
+
+      result.isSuccess === true
+    }
+  }
+
+  "Email Validation in Dummy Form" should {
+
+    "return an error when the email field is empty" in {
+
+      val valueToTest = ""
+      val editableField = mockEditText(valueToTest)
+
+      val form = FormModel(None, Some(editableField), None)
+
+      val result = isValidEmailField(form)
+
+      result.isSuccess === false
+      result.toEither.left.get.details === s"email can not be empty."
+    }
+
+    "return success when an invalid email is specified" in {
+
+      val valueToTest = "invalid.com"
+      val editableField = mockEditText(valueToTest)
+
+      val form = FormModel(None, Some(editableField), None)
+
+      val result = isValidEmailField(form)
+
+      result.isSuccess === false
+      result.toEither.left.get.details === s"$valueToTest is not a valid email address."
+    }
+
+    "return success when a valid email is specified" in {
+
+      val valueToTest = "johndoe@domain.com"
+      val editableField = mockEditText(valueToTest)
+
+      val form = FormModel(None, Some(editableField), None)
+
+      val result = isValidEmailField(form)
+
+      result.isSuccess === true
+    }
+  }
+
+  "Age Validation in Dummy Form" should {
+
+    "return an error when the age field is empty" in {
+
+      val valueToTest = ""
+      val editableField = mockEditText(valueToTest)
+
+      val form = FormModel(None, None, Some(editableField))
+
+      val result = isValidAgeField(form)
+
+      result.isSuccess === false
+      result.toEither.left.get.details === s"age can not be empty."
+    }
+
+    "return success when an invalid age is specified" in {
+
+      val valueToTest = "AA"
+      val editableField = mockEditText(valueToTest)
+
+      val form = FormModel(None, None, Some(editableField))
+
+      val result = isValidAgeField(form)
+
+      result.isSuccess === false
+      result.toEither.left.get.details === s"$valueToTest is an invalid number."
+    }
+
+    "return success when an invalid age (float number) is specified" in {
+
+      val valueToTest = "4.4"
+      val editableField = mockEditText(valueToTest)
+
+      val form = FormModel(None, None, Some(editableField))
+
+      val result = isValidAgeField(form)
+
+      result.isSuccess === false
+      result.toEither.left.get.details === s"$valueToTest is an invalid number."
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Usage of get on optional type.](https://www.codacy.com/app/47Degrees/scala-android/file/661377200/issues/source?bid=2339068&fileBranchId=2339108#l121)